src/api/FileCollaborators.js

```diff
@@ -24,26 +30,34 @@ class FileCollaborators extends MarkerBasedAPI {
     }
 
     /**
-     * Generic success handler
+     * Transform result of API response
      *
      * @param {Object} data the response data
      */
-    successHandler = (data: any): void => {
+    successHandler = (data: CollaboratorsAPIResponse): void => {
         if (this.isDestroyed() || typeof this.successCallback !== 'function') {
             return;
         }
 
-        const { entries } = data;
-        const collaborators = entries.map(collab => {
-            const { id, name, login } = collab;
-            return {
-                id,
-                name,
-                item: { id, name, email: login },
-            };
-        });
+        // Transform into "mention selector" format:
+        const collaboratorSelectorItems: SelectorItems<UserMini | GroupMini> = data.entries.map(
+            (collab: UserMini | GroupMini) => {
+                let item;
+                if (collab.type === 'group') {
+                    item = collab; // flow needs assignment to happen after type refinement
```

Comment:
To confirm, this means that the following wouldn't work?

```
const item = { ...collab };

if (item.type !== 'group') {
    item.email = item.login;
}

return {
    id: item.id,
    name: item.name,
    item,
};

Reply:
Correct.

```
Cannot assign `item.login` to `item.email` because property `email` is missing in  object literal [1].Flow(InferError)
```